### PR TITLE
Expression profile in core

### LINKED ISSUE #1732: Serialization for Expression Profile domain object
See #1731

### LINKED ISSUE #1809: Serializing Expression Profile to pkml
For this to succeed the following steps are necessary:

1) copy the expression profile domain object that has already been created in MoBI (is supposed to end up in Core anyway - we should discuss how we handle this through the development process though).

2) create a mapper to convert the PK-Sim expression profile to Core expression profile.

3) create a serializer to pkml for the expression profile to extend the pkml persistor.

### PR DESCRIPTION
**Pull request by georgeDaskalakis**
closes #1732
closes #1809

### CODE REVIEW

File: src/OSPSuite.Core/Domain/Builder/ExpressionTypes.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections.Generic;
```
Comment (rwmcintosh) [inline src/OSPSuite.Core/Domain/Builder/ExpressionTypes.cs:1]: Some new code here to handle the ExpressionType serialization

File: src/OSPSuite.Core/Serialization/Xml/BuildingBlockXmlSerializer.cs [hunk 2]
```diff
@@ -98,6 +98,15 @@ public override void PerformMapping()
       }
    }
 
+   public class ExpressionProfileBuildingBlockXmlSerializer : StartValuesBuildingBlockXmlSerializer<ExpressionProfileBuildingBlock, ExpressionParameter>
```
Comment (rwmcintosh) [inline src/OSPSuite.Core/Serialization/Xml/BuildingBlockXmlSerializer.cs:101]: New code for serialization of core objects

File: src/OSPSuite.Core/Serialization/Xml/ExpressionParameterXmlSerializer.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+using OSPSuite.Core.Domain.Builder;
```
Comment (rwmcintosh) [inline src/OSPSuite.Core/Serialization/Xml/ExpressionParameterXmlSerializer.cs:1]: New code for serialization of core objects

File: src/OSPSuite.Core/Serialization/Xml/ExpressionTypeXmlAttributeMapper.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+using OSPSuite.Core.Domain.Builder;
```
Comment (rwmcintosh) [inline src/OSPSuite.Core/Serialization/Xml/ExpressionTypeXmlAttributeMapper.cs:1]: New code for serialization of core objects

File: src/OSPSuite.Core/Serialization/Xml/OSPSuiteXmlSerializerRepository.cs [hunk 5]
```diff
@@ -58,8 +58,9 @@ protected override void AddInitialMappers()
          AttributeMapperRepository.AddAttributeMapper(new ValueOriginSourceXmlAttributeMapper<SerializationContext>());
          AttributeMapperRepository.AddAttributeMapper(new ValueOriginDeterminationMethodXmlAttributeMapper<SerializationContext>());
          AttributeMapperRepository.AddAttributeMapper(new RunStatusXmlAttributeMapper());
+         AttributeMapperRepository.AddAttributeMapper(new ExpressionTypeXmlAttributeMapper());
```
Comment (rwmcintosh) [inline src/OSPSuite.Core/Serialization/Xml/OSPSuiteXmlSerializerRepository.cs:61]: New code for serialization of core objects

File: src/OSPSuite.Core/Serialization/Xml/PathAndValueEntityXmlSerializer.cs [hunk 6]
```diff
@@ -0,0 +1,15 @@
+using OSPSuite.Core.Domain.Builder;
```
Comment (rwmcintosh) [inline src/OSPSuite.Core/Serialization/Xml/PathAndValueEntityXmlSerializer.cs:1]: New code for serialization of core objects

File: src/OSPSuite.Core/Domain/Builder/ExpressionParameter.cs [hunk 7]
```diff
@@ -0,0 +1,22 @@
+﻿using OSPSuite.Core.Domain.Services;
+
+namespace OSPSuite.Core.Domain.Builder
+{
+   public class ExpressionParameter : PathAndValueEntity, IStartValue
+   {
+      /// <summary>
+      ///    Do not use! When refactoring on promotion to core, this should be removed
+      /// </summary>
+      public double? StartValue { get; set; }
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/Builder/ExpressionParameter.cs:10]: Should this be removed?

File: src/OSPSuite.Core/Domain/Builder/ExpressionProfileBuildingBlock.cs [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+      public ExpressionType Type { set; get; }
+
+      public int PKSimVersion { set; get; }
+
+      public virtual string Category { get; private set; }
+
+      public override string Name
+      {
+         get => PKSimCoreConstants.ContainerName.ExpressionProfileName(MoleculeName, Species, Category);
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/Builder/ExpressionProfileBuildingBlock.cs:21]: PKSimCoreConstants...not sure about this naming. Can we move this to the usual constant (and also use it in PKSim?)

File: src/OSPSuite.Core/Domain/Builder/ExpressionProfileBuildingBlock.cs [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+      public override void UpdatePropertiesFrom(IUpdatable source, ICloneManager cloneManager)
+      {
+         base.UpdatePropertiesFrom(source, cloneManager);
+         var sourceExpressionProfile = source as ExpressionProfileBuildingBlock;
+         if (sourceExpressionProfile == null) return;
+
+         Type = sourceExpressionProfile.Type;
+         PKSimVersion = sourceExpressionProfile.PKSimVersion;
+         Name = sourceExpressionProfile.Name;
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/Builder/ExpressionProfileBuildingBlock.cs:47]: Why is this required? Isn't done by the base. method?

File: src/OSPSuite.Core/Comparison/ExpressionProfileDiffBuilder.cs [hunk 10]
```diff
@@ -0,0 +1,19 @@
+﻿using OSPSuite.Core.Domain.Builder;
+
+namespace OSPSuite.Core.Comparison
+{
+   public class ExpressionProfileBuildingBlockDiffBuilder : StartValueBuildingBlockDiffBuilder<ExpressionParameter>
+   {
+      public ExpressionProfileBuildingBlockDiffBuilder(ObjectBaseDiffBuilder objectBaseDiffBuilder, EnumerableComparer enumerableComparer) : base(objectBaseDiffBuilder, enumerableComparer)
```
Comment (msevestre) [inline src/OSPSuite.Core/Comparison/ExpressionProfileDiffBuilder.cs:7]: I think some properties are missing.
Type, PKSImVersion etc

Comment (msevestre) [reply src/OSPSuite.Core/Comparison/ExpressionProfileDiffBuilder.cs:7]: In short, evey single propoerty should be compared, based on the settings of the comparison

File: src/OSPSuite.Core/Domain/Builder/ExpressionTypes.cs [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+{
+
+   public enum ExpressionTypesId
+   {
+      Transport,
+      Enzyme,
+      BindingPartner
+   }
+
+   public static class TempAssets
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/Builder/ExpressionTypes.cs:14]: ??

File: src/OSPSuite.Core/Domain/Builder/ExpressionTypes.cs [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+         return _types.First(x => Equals(x.Id, expressionTypeId));
+      }
+   }
+
+   public class ExpressionType
+   {
+      public ExpressionTypesId Id { get; }
+      public string IconName { get; }
+      public string DisplayName { get; }
+      public ExpressionType(string iconName, string displayName, ExpressionTypesId expressionTypesId)
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/Builder/ExpressionTypes.cs:58]: Let's change the order and put the id first. It's weird to initialize an object by the icon

File: src/OSPSuite.Core/Domain/Builder/ExpressionParameter.cs (hunk 7 above)
Comment (rwmcintosh) [reply src/OSPSuite.Core/Domain/Builder/ExpressionParameter.cs:10]: The very next issue after this PR is going to be refactoring all the StartValues -> EntityWithPath to make sure it makes sense.

I had this code with this comment at one point, but then we removed the PathWithEntity.Value and just had StartValue to eliminate confusion.

For this PR, I would not bother to take the comment out.

File: src/OSPSuite.Core/PKSimCoreConstants.cs [hunk 13]
```diff
@@ -0,0 +1,47 @@
+﻿using OSPSuite.Core.Domain;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using OSPSuite.Utility.Extensions;
+
+namespace OSPSuite.Core
+{
+   public static class PKSimCoreConstants
```
Comment (msevestre) [inline src/OSPSuite.Core/PKSimCoreConstants.cs:10]: yep let's also remove this class

File: tests/OSPSuite.Core.IntegrationTests/Serializers/ExpressionParameterXmlSerializerSpecs.cs [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Core.Domain.Formulas;
+using OSPSuite.Core.Helpers;
+
+namespace OSPSuite.Core.Serializers
+{
+   public class ExpressionParameterXmlSerializerSpecs : ModellingXmlSerializerBaseSpecs
+   {
+      [Test]
```
Comment (msevestre) [inline tests/OSPSuite.Core.IntegrationTests/Serializers/ExpressionParameterXmlSerializerSpecs.cs:11]: Observation

File: tests/OSPSuite.Core.IntegrationTests/Serializers/ExpressionParameterXmlSerializerSpecs.cs [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+      public void TestSerialization()
+      {
+         IFormula formula = CreateObject<ExplicitFormula>().WithName("F.Erika").WithDimension(DimensionLength).WithFormulaString("A * 2");
```
Comment (msevestre) [inline tests/OSPSuite.Core.IntegrationTests/Serializers/ExpressionParameterXmlSerializerSpecs.cs:14]: var

File: tests/OSPSuite.Core.IntegrationTests/Serializers/ExpressionParameterXmlSerializerSpecs.cs [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+         IFormulaUsablePath fup = new FormulaUsablePath(new[] { "aa", "bb" }).WithAlias("b").WithDimension(DimensionLength);
```
Comment (msevestre) [inline tests/OSPSuite.Core.IntegrationTests/Serializers/ExpressionParameterXmlSerializerSpecs.cs:15]: var

File: tests/OSPSuite.Core.Tests/Domain/ExpressionParameterSpecs.cs [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+   public abstract class expression_parameter_equivalency_should_test_negative : when_testing_equivalency_in_expression_parameters
+   {
+      [Observation]
+      public void should_not_be_equivalent()
+      {
+         _result.ShouldBeFalse();
+      }
+   }
+
+   public class when_testing_expression_parameters_with_different_scale_factor : equivalency_should_test_negative
```
Comment (msevestre) [inline tests/OSPSuite.Core.Tests/Domain/ExpressionParameterSpecs.cs:97]: We usually have When and not when

**Review by msevestre — CHANGES_REQUESTED** [review]

**Comment by rwmcintosh** [comment]
Everything addressed now. I'm going to update the app PR's

File: src/OSPSuite.Core/Comparison/BuildingBlockDiffBuilder.cs [hunk 18]
```diff
@@ -24,60 +24,60 @@ protected BuildingBlockDiffBuilder(ObjectBaseDiffBuilder objectBaseDiffBuilder,
...
-   public class ReactionBuildingBlockDiffBuilder : BuildingBlockDiffBuilder<IReactionBuilder>
+   public class ReactionBuildingBlockDiffBuilder : BuildingBlockDiffBuilder<IReactionBuildingBlock, IReactionBuilder>
    {
       public ReactionBuildingBlockDiffBuilder(ObjectBaseDiffBuilder objectBaseDiffBuilder, EnumerableComparer enumerableComparer) : base(objectBaseDiffBuilder, enumerableComparer)
       {
       }
    }
 
-   public class PassiveTransportBuildingBlockDiffBuilder : BuildingBlockDiffBuilder<ITransportBuilder>
+   public class PassiveTransportBuildingBlockDiffBuilder : BuildingBlockDiffBuilder<PassiveTransportBuildingBlock, ITransportBuilder>
```
Comment (msevestre) [inline src/OSPSuite.Core/Comparison/BuildingBlockDiffBuilder.cs:56]: I am bit confused here. Sometimes, the TBuildingBlock in an interface (line 49), sometimes it's an actual implementation. Why isn't it consistent?

File: src/OSPSuite.Assets/UIConstants.cs [hunk 19]
```diff
@@ -18,6 +18,11 @@ public static class SizeAndLocation
 
    public static class Captions
    {
+      public static string Transporter = "Transporter";
```
Comment (msevestre) [inline src/OSPSuite.Assets/UIConstants.cs:21]: Why are those guys static strign where everything under is readonly string? Should we make this consistent?

File: src/OSPSuite.Core/Comparison/ExpressionProfileDiffBuilder.cs [hunk 20]
```diff
@@ -0,0 +1,30 @@
...
+      {
+
+      }
+
+      public override void Compare(IComparison<ExpressionProfileBuildingBlock> comparison)
+      {
+         base.Compare(comparison);
+         CompareValues(x => x.PKSimVersion, x => x.PKSimVersion, comparison);
+         CompareStringValues(x => x.MoleculeName, x => x.MoleculeName, comparison);
```
Comment (msevestre) [inline src/OSPSuite.Core/Comparison/ExpressionProfileDiffBuilder.cs:16]: Molecule Name, Species, Category are all derived from the name so it's probably not required because it will be redundant. Name is also probably compared in the base class
Last there is a settings that is used. (compare relevant properties or whatever). PKSim Version is probably not one of them. Type ? not sure

File: src/OSPSuite.Core/Comparison/ExpressionProfileDiffBuilder.cs [hunk 21]
```diff
@@ -0,0 +1,30 @@
...
+         CompareStringValues(x => x.Species, x => x.Species, comparison);
+         CompareStringValues(x => x.Category, x => x.Category, comparison);
+         CompareStringValues(x => x.Name, x => x.Name, comparison);
+         CompareValues(x => x.Type, x => x.Type, comparison, (x,y) => Equals(x.DisplayName, y.DisplayName), (x,y) => y.DisplayName);
```
Comment (msevestre) [inline src/OSPSuite.Core/Comparison/ExpressionProfileDiffBuilder.cs:20]: why the overlaod of equals here? Type is a static instance correct? They should be the same. Also if you override ToString in the type,e you do not need the DisplayName in the c comparison either

File: src/OSPSuite.Core/Domain/Builder/ExpressionParameter.cs (hunk 7 above)
Comment (msevestre) [reply src/OSPSuite.Core/Domain/Builder/ExpressionParameter.cs:10]: ok

File: src/OSPSuite.Core/Domain/Builder/ExpressionProfileBuildingBlock.cs (hunk 9 above)
Comment (msevestre) [reply src/OSPSuite.Core/Domain/Builder/ExpressionProfileBuildingBlock.cs:48]: Speaking about the Name here. I don't think we need this

File: src/OSPSuite.Core/Domain/Builder/ExpressionTypes.cs [hunk 22]
```diff
@@ -0,0 +1,45 @@
...
+         return _types.First(x => Equals(x.Id, expressionTypeId));
+      }
+   }
+
+   public class ExpressionType
+   {
+      public ExpressionTypesId Id { get; }
+      public string IconName { get; }
+      public string DisplayName { get; }
+      public ExpressionType(ExpressionTypesId expressionTypesId, string iconName, string displayName)
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/Builder/ExpressionTypes.cs:38]: here overrride ToString to return DisplayName. That way the debugger will also show you the right  info

File: src/OSPSuite.Core/Domain/Builder/ExpressionTypes.cs [hunk 23]
```diff
@@ -0,0 +1,45 @@
...
+      private static readonly List<ExpressionType> _types = new List<ExpressionType>
+      {
+         TransportProtein,
+         MetabolizingEnzyme,
+         ProteinBindingPartner
+      };
+
+      public static ExpressionType ById(ExpressionTypesId expressionTypeId)
+      {
+         return _types.First(x => Equals(x.Id, expressionTypeId));
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/Builder/ExpressionTypes.cs:29]: what about you make it a cache by id here so that you retrieve by id? It seems more idiomatic

File: tests/OSPSuite.Core.Tests/Domain/ExpressionProfileBuildingBlockSpecs.cs [hunk 24]
```diff
@@ -0,0 +1,91 @@
...
+         sut.Name.ShouldBeEqualTo("Molecule|Species|Name");
+         sut.Type.ShouldBeEqualTo(ExpressionTypes.MetabolizingEnzyme);
+         sut.PKSimVersion.ShouldBeEqualTo(11);
+         sut.Count().ShouldBeEqualTo(1);
+      }
+   }
+
+   //commenting out, as long as we do not have an icon based on type
+   /*
+   public class when_reading_the_icon_name_for_the_building_block : concern_for_ExpressionProfileBuildingBlock
```
Comment (msevestre) [inline tests/OSPSuite.Core.Tests/Domain/ExpressionProfileBuildingBlockSpecs.cs:60]: why is this still commented out? There is the icons defined in the type right?

File: src/OSPSuite.Assets/UIConstants.cs (hunk 19 above)
Comment (georgeDaskalakis) [reply src/OSPSuite.Assets/UIConstants.cs:21]: done

File: src/OSPSuite.Core/Domain/Builder/ExpressionProfileBuildingBlock.cs (hunk 9 above)
Comment (georgeDaskalakis) [reply src/OSPSuite.Core/Domain/Builder/ExpressionProfileBuildingBlock.cs:48]: Yeap, it is done in ObjectBase.cs 
removed

File: src/OSPSuite.Core/Comparison/ExpressionProfileDiffBuilder.cs (hunk 10 above)
Comment (georgeDaskalakis) [reply src/OSPSuite.Core/Comparison/ExpressionProfileDiffBuilder.cs:7]: fixed

File: src/OSPSuite.Core/Domain/Builder/ExpressionTypes.cs (hunk 12 above)
Comment (georgeDaskalakis) [reply src/OSPSuite.Core/Domain/Builder/ExpressionTypes.cs]: done

File: src/OSPSuite.Core/Domain/Builder/ExpressionTypes.cs (hunk 11 above)
Comment (georgeDaskalakis) [reply src/OSPSuite.Core/Domain/Builder/ExpressionTypes.cs]: removed

File: src/OSPSuite.Core/PKSimCoreConstants.cs (hunk 13 above)
Comment (georgeDaskalakis) [reply src/OSPSuite.Core/PKSimCoreConstants.cs]: done

File: tests/OSPSuite.Core.IntegrationTests/Serializers/ExpressionParameterXmlSerializerSpecs.cs (hunk 14 above)
Comment (georgeDaskalakis) [reply tests/OSPSuite.Core.IntegrationTests/Serializers/ExpressionParameterXmlSerializerSpecs.cs]: done

File: tests/OSPSuite.Core.IntegrationTests/Serializers/ExpressionParameterXmlSerializerSpecs.cs (hunk 15 above)
Comment (georgeDaskalakis) [reply tests/OSPSuite.Core.IntegrationTests/Serializers/ExpressionParameterXmlSerializerSpecs.cs]: done

File: tests/OSPSuite.Core.IntegrationTests/Serializers/ExpressionParameterXmlSerializerSpecs.cs (hunk 16 above)
Comment (georgeDaskalakis) [reply tests/OSPSuite.Core.IntegrationTests/Serializers/ExpressionParameterXmlSerializerSpecs.cs]: done

File: tests/OSPSuite.Core.Tests/Domain/ExpressionParameterSpecs.cs (hunk 17 above)
Comment (georgeDaskalakis) [reply tests/OSPSuite.Core.Tests/Domain/ExpressionParameterSpecs.cs]: done

File: src/OSPSuite.Core/Comparison/BuildingBlockDiffBuilder.cs (hunk 18 above)
Comment (georgeDaskalakis) [reply src/OSPSuite.Core/Comparison/BuildingBlockDiffBuilder.cs:56]: changerd to consistently use the interfaces

File: src/OSPSuite.Core/Comparison/ExpressionProfileDiffBuilder.cs (hunk 20 above)
Comment (georgeDaskalakis) [reply src/OSPSuite.Core/Comparison/ExpressionProfileDiffBuilder.cs:16]: removed Molecule Name, Species, Category and Name, will test the comparison to see that everything still works.
@msevestre I am not sure wha tyou mean by the last part:

> Molecule Name, Species, Category are all derived from the name so it's probably not required because it will be redundant. Name is also probably compared in the base class
Last there is a settings that is used. (compare relevant properties or whatever). PKSim Version is probably not one of them. Type ? not sure

File: src/OSPSuite.Core/Comparison/ExpressionProfileDiffBuilder.cs (hunk 21 above)
Comment (georgeDaskalakis) [reply src/OSPSuite.Core/Comparison/ExpressionProfileDiffBuilder.cs:20]: @msevestre hehe possibly this is what you menat above? Removing it, again I will test the new outcome of the comparison

File: tests/OSPSuite.Core.Tests/Domain/ExpressionProfileBuildingBlockSpecs.cs (hunk 24 above)
Comment (georgeDaskalakis) [reply tests/OSPSuite.Core.Tests/Domain/ExpressionProfileBuildingBlockSpecs.cs:60]: added back in

File: src/OSPSuite.Core/Domain/Builder/ExpressionTypes.cs (hunk 22 above)
Comment (georgeDaskalakis) [reply src/OSPSuite.Core/Domain/Builder/ExpressionTypes.cs:38]: done

File: src/OSPSuite.Core/Domain/Builder/ExpressionTypes.cs (hunk 23 above)
Comment (georgeDaskalakis) [reply src/OSPSuite.Core/Domain/Builder/ExpressionTypes.cs:29]: done

**Comment by georgeDaskalakis** [comment]
@msevestre apart form the unresolved comment above ( which could even be fixed by the fix of the next comment), this is ready.

File: src/OSPSuite.Core/Comparison/ExpressionProfileDiffBuilder.cs (hunk 21 above)
Comment (rwmcintosh) [reply src/OSPSuite.Core/Comparison/ExpressionProfileDiffBuilder.cs]: Type is an 'ExpressionType' That means Enzyme, Transporter, BindingPartner

Comment (rwmcintosh) [reply src/OSPSuite.Core/Comparison/ExpressionProfileDiffBuilder.cs]: We need that comparison put back in if you took it out, but if you want, rename the property `ExpressionType` instead of `Type` to avoid confusion.

Comment (rwmcintosh) [reply src/OSPSuite.Core/Comparison/ExpressionProfileDiffBuilder.cs]: Ok, after thinking about it more, I get what you are saying. This `Type` is just doing a reference compare anyway, so why override, that was because I needed the DisplayName in the diff. Your second sentence is override ToString(), which solves that. So, indeed, we don't need this line back in.

File: src/OSPSuite.Core/Domain/Builder/ExpressionProfileBuildingBlock.cs (hunk 9 above)
Comment (rwmcintosh) [reply src/OSPSuite.Core/Domain/Builder/ExpressionProfileBuildingBlock.cs]: To use that base method name clone we would need to change how we store this stuff. We allow name to be set, but it gets decomposed and stored in the other 3 properties. When recalled, we build the name from those properties again.

If we want to use the ObjectBase name clone, then we need to reverse that storage/recall strategy to store all the properties in the _name, then decompose when recalled for Species etc.

```
      public virtual void UpdatePropertiesFrom(IUpdatable source, ICloneManager cloneManager)
      {
         var sourceObjectBase = source as IObjectBase;
         if (sourceObjectBase == null) return;

         //ID should not be updated since it should have been defined in constructor or during the object construction process
         _name = sourceObjectBase.Name;
         Icon = sourceObjectBase.Icon;
         Description = sourceObjectBase.Description;
      }
```

Comment (rwmcintosh) [reply src/OSPSuite.Core/Domain/Builder/ExpressionProfileBuildingBlock.cs]: or put `Name = sourceObjectBase.Name;` instead of `_name = sourceObjectBase.Name;`

Comment (georgeDaskalakis) [reply src/OSPSuite.Core/Domain/Builder/ExpressionProfileBuildingBlock.cs]: We are going to commit a change with the Name being assigne to in UpdatePropertiesFrom override - changing the objectBase seems a bit dangerous

File: src/OSPSuite.Core/Comparison/ExpressionProfileDiffBuilder.cs (hunk 10 above)
Comment (msevestre) [reply src/OSPSuite.Core/Comparison/ExpressionProfileDiffBuilder.cs:7]: so the settings of the comparison are not used./
See ParameterDiffBuilder

File: src/OSPSuite.Core/Domain/Builder/ExpressionProfileBuildingBlock.cs [hunk 25]
```diff
@@ -0,0 +1,51 @@
...
+
+      public string Species { get; private set; }
+
+      public ExpressionType Type { set; get; }
+
+      public int PKSimVersion { set; get; }
+
+      public virtual string Category { get; private set; }
+
+      public override string Name
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/Builder/ExpressionProfileBuildingBlock.cs:20]: Looking at the maching PKSIm implementation , events are missing here
You need to throw ent OnPRopertyChangedEvent for Name

File: src/OSPSuite.Core/Domain/Builder/ExpressionProfileBuildingBlock.cs (hunk 9 above)
Comment (msevestre) [reply src/OSPSuite.Core/Domain/Builder/ExpressionProfileBuildingBlock.cs:48]: Let's add a comment why it's required then. THis just felt wrong
And ObjectBase should probably update Name to be honest. This looks wrong to me

**Review by msevestre — CHANGES_REQUESTED** [review]
Looks good to me. Only issue is with the NotifyPropertyChanged missing for name

**Comment by rwmcintosh** [comment]
I think that's everything now.

**Review by msevestre — APPROVED** [review]
Still don't see that the comparison is implemented using the configurations settings but who cares
